barretenberg/cpp/src/barretenberg/vm/avm/trace/trace.cpp

```diff
@@ -134,6 +136,56 @@ bool check_tag_integral(AvmMemoryTag tag)
  *                                   HELPERS
  **************************************************************************************************/
 
+void AvmTraceBuilder::checkpoint_non_revertible_state()
+{
+    merkle_tree_trace_builder.checkpoint_non_revertible_state();
+}
+void AvmTraceBuilder::rollback_to_non_revertible_checkpoint()
+{
+    merkle_tree_trace_builder.rollback_to_non_revertible_checkpoint();
+}
+
+std::vector<uint8_t> AvmTraceBuilder::get_bytecode(const FF contract_address, bool check_membership)
+{
+    // uint32_t clk = 0;
+    // auto clk = static_cast<uint32_t>(main_trace.size()) + 1;
```

Comment:
This is needed by the merkle call we'll use eventually (`perform_nullifier_read`). But what should clk be? `get_bytecode` isn't an instruction, it's done at the start of each enqueued and nested call.